tdms/include/input_flags.h

```diff
@@ -0,0 +1,142 @@
+enum SolverMethod : bool { PseudoSpectral = true, FiniteDifference = false };
+//! The interpolation method to use when extracting field values at Yee
+//! cell centres. use_bli is the corresponding flag.
+enum InterpolationMethod : bool { BandLimited = true, Cubic = false };
+}// namespace tdms_flags
+
+class InputFlags {
+private:
+  //! Name of the input file to read flags from
+  std::string input_filename = "input_file.mat";
+  //! Flag values present in the input file
+  bool flag_values[NFLAGS] = {false};
+
+  /**
+   * @brief Returns the internal index in flag_values of the flag with the name
+   * provided.
+   *
+   * @param flag_name Name of the flag
+   * @return const int Index of this flag's value in flag_values
+   */
+  const int position_from_name(const std::string &flag_name) const {
+    auto position = std::find(tdms_flags::flag_variables.begin(),
+                              tdms_flags::flag_variables.end(), flag_name);
+    if (position == tdms_flags::flag_variables.end()) {
+      // Could not find the flag in the list of expected flags, return an error
+      throw std::runtime_error(flag_name + " is not an expected flag");
+    }
+    return std::distance(tdms_flags::flag_variables.begin(), position);
+  }
+
+  /**
+   * @brief Fetch the value of the flag variable from the input file.
+   *
+   * @param flag_name The name of the flag to fetch
+   * @param fail_on_not_found If true, throw an error if the flag cannot be
+   * located. Otherwise, return the default flag value.
+   * @return true This flag was present and set to true
+   * @return false This flag was present and set to false, OR was not present
+   */
+  bool fetch_flag_value(const std::string flag_name,
+                        bool fail_on_not_found = false) const {
+    MATFile *input_file = matOpen(input_filename.c_str(), "r");
+    bool return_value;
+
+    mxArray *ptr_to_flag = matGetVariable(input_file, flag_name.c_str());
+    if (ptr_to_flag == nullptr) {
+      // This flag was not present in the input file.
+      // We either return the default value (0, flag not present), or we throw
+      // an error if fail_on_not_found is set to true.
+      if (fail_on_not_found) {
+        throw std::runtime_error(flag_name + " was not present in " +
+                                 input_filename);
+      } else {
+        return_value = false;
+      }
+    } else if (mxIsLogicalScalar(ptr_to_flag)) {
+      // The flag is present in the input file, and is a scalar boolean.
+      // Return its value
+      return_value = (bool) *mxGetPr(ptr_to_flag);
+    } else {
+      // The flag appears to be present, but is not scalar-valued. Throw
+      // exception.
+      throw std::runtime_error(flag_name + " is present in " + input_filename +
+                               ", but is not scalar.");
+    }
+
+    // Cleanup MATLAB structs that are still in memory
+    matClose(input_file);
+    mxDestroyArray(ptr_to_flag);
+    return return_value;
+  }
```

Comment:
You're doing quite a lot of work here in the header, I'd put it in the implementation.